src/java/org/apache/cassandra/index/sai/view/IndexViewManager.java

```diff
@@ -63,12 +62,14 @@ public IndexViewManager(IndexContext context)
     IndexViewManager(IndexContext context, Collection<Descriptor> sstables, Collection<SSTableIndex> indices)
     {
         this.context = context;
-        this.view.set(new View(context, sstables, indices));
+        this.viewRef.set(new View(context, sstables, indices));
+        // View references the indices, so we release them here.
+        indices.forEach(SSTableIndex::release);
```

Comment:
I don't think it's right to release the reference for provided `indices`. The contract is that `View` will hold an extra reference to prevent `SSTableIndex` being released by sstable obsoletion.

Reply:
Also in production, it always uses the `IndexViewManager` constructor with empty sstable indexes